File: app/src/main/java/app/revanced/bilibili/meta/CookieInfo.java
package app.revanced.bilibili.meta;

import com.alibaba.fastjson.annotation.JSONField;

import java.util.List;

/* compiled from: BL */
/* loaded from: classes5.dex */
public class CookieInfo {
    @JSONField(name = "cookies")
    public List<CookieBean> cookies;

    public CookieInfo(List<CookieBean> cookies) {
        this.cookies = cookies;
    }

    /* compiled from: BL */
    /* loaded from: classes5.dex */
    public static class CookieBean {
        @JSONField(name = "name")
        public String name;
        @JSONField(name = "value")
        public String value;

        public CookieBean(String name, String value) {
            this.name = name;
            this.value = value;
        }
    }
}

File: integrations/app/src/main/java/app/revanced/bilibili/account/model/CookieInfo.kt
package app.revanced.bilibili.account.model

import app.revanced.bilibili.model.serializer.IntAsBoolean
import app.revanced.bilibili.model.serializer.SecondsAsDate
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.Date

@Serializable
class CookieInfo(
    @SerialName("cookies")
    val cookies: List<CookieBean> = listOf(),
    @SerialName("domains")
    val domains: List<String> = listOf(),
) {
    @Serializable
    class CookieBean(
        @SerialName("expires")
        val expires: SecondsAsDate = Date(),
        @SerialName("http_only")
        val httpOnly: IntAsBoolean = false,
        @SerialName("name")
        val name: String = "",
        @SerialName("secure")
        val secure: IntAsBoolean = false,
        @SerialName("value")
        val value: String = ""
    )
}
